docs/examples/download_file.py

```diff
@@ -1,13 +1,31 @@
+import io
+import functools
+
 from twisted.internet.task import react
+from twisted.internet import threads, defer
 
 import treq
 
 
-def download_file(reactor, url, destination_filename):
-    destination = open(destination_filename, 'wb')
-    d = treq.get(url, unbuffered=True)
-    d.addCallback(treq.collect, destination.write)
-    d.addBoth(lambda _: destination.close())
-    return d
+def deferToThread(reactor, f, *args, **kwargs):
```

Comment:
I think for example code it would make more sense to call `deferToThreadPool` inline?